docs/tapiro/introduction.md

```diff
@@ -0,0 +1,85 @@
+---
+id: introduction
+title: Introduction
+---
+
+Tapiro parses your Scala controllers to generate HTTP endpoints.
+
+A Scala controller is a trait defined as follows:
+
+```scala mdoc
+import scala.annotation.StaticAnnotation
+
+class query extends StaticAnnotation
+class command extends StaticAnnotation
+
+case class Cat(name: String)
+case class Error(msg: String)
+
+trait Cats[F[_], AuthToken] {
+  @query //translate this to a GET
+  def findCutestCat(): F[Either[Error, Cat]]
+
+  @command //translate this to a POST
+  def doSomethingWithTheCat(catId: Int, token: AuthToken): F[Either[Error, Unit]]
+}
+```
+
+For each controller tapiro generates two files:
+- `CatsEndpoints.scala` containing the HTTP api description using https://tapir-scala.readthedocs.io/
+- `CatsHttp4sEndpoints.scala` or `CatsAkkaHttpEndpoints.scala` depeneding on the HTTP server the user is using.
+
+The resulting routes can be simply used like this (http4s example):
+```scala
```

Comment:
this block is not typechecked

Reply:
☝️ farei un esempio più completo che effettivamente compili

Reply:
Per avere esempio completo che compili servono dipendenze, per questo stavo rimandando la cosa.

Reply:
Aggiunto superesempione.

Reply:
@gabro pingone